src/net.cpp

```diff
@@ -682,6 +691,19 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes)
         nBytes -= handled;
 
         if (msg.complete()) {
+
+            //store received bytes per command
+            //to prevent a memory DOS, only allow valid commands
+            std::string command(msg.hdr.pchCommand);
+            for (unsigned int i = 0; i < sizeof(logAllowIncomingCmds)/sizeof(logAllowIncomingCmds[0]); i++)
```

Comment:
I've thought of - what I think is an even better solution - as it doesn't require recourse to anything besides the map itself: prepopulate the `mapRecvBytesPerCmd`. At initialization:
```
    for (unsigned int i = 0; i < sizeof(logAllowIncomingCmds)/sizeof(logAllowIncomingCmds[0]); i++)
        mapRecvBytesPerCmd[logAllowIncomingCmds[i]] = 0;
```
Then only add if the item exists, e.g.
```c++
    mapCmdSize_t::iterator i = mapRecvBytesPerCmd.find(msg.hdr.pchCommand);
    if (i != mapRecvBytesPerCmd.end())
        i->second += 1;
```